### ISSUE TITLE
Updating Hardware on Pi fails 

### ISSUE THREAD

**Issue by APLSKO** [kihosk/kiosk-raspberry#67]
I tried to install kiosk on Pi3+. 
Image setup worked, Pi is booting up and starting the update process. 
It seems like the download is still working, but then the update hangs. 
I tried it with v. 0.0.15 and 0.0.14 - same topic.
Is there an known issue?

**Comment by Herbiland** [comment]
Same issue here with Pi4

**Comment by Postiopolis** [comment]
Same here on Raspberry Pi4 . I've tried 15 , 14 and 13 all with the same issue. 1 hour of downloading updates seems a bit much. Or am I wrong?

**Comment by Herbiland** [comment]
Anyone any solution guys? I have tryed almost everything. But i wasnt succesfull

**Comment by Postiopolis** [comment]
I left mine on "update" all weekend and came back today and it was still saying the same thing. I guess I'll have to try something else beside Kiosk. :(

**Comment by marcusburge** [comment]
Likewise. Perhaps a bug in a URL called by the update process, perhaps it's timing out?

**Comment by arismoratalla** [comment]
Same here. I left it overnight and it's still on Updating Hardware... It seems connected to the internet though coz it gives a message when I disconnect the ethernet cable.

**Comment by hutje** [comment]
Same with me. RPi3 and RPi4...

**Comment by igolus** [comment]
Hello same for me with RPPI4, I tried also with ethernet cable with no success

**Comment by DecRodgers** [comment]
Same here with 0.15, got a 4GB Pi with Ethernet, I get the initial 'KIOSK - Setting up' screen, reboots, then gets stuck on 'UPDATING HARDWARE...". I've left it for 6+hours and reflashed the image with no change.
When I take the ethernet out, it detects that. It returns to 'UPDATING HARDWARE' when plugged back in.

**Comment by igolus** [comment]
Finding some logs:

[2021-10-21 13:44:15.738] [info] Electron App: New version 0.1.89 has been downloaded to /home/pi/.cache/kiosk-updater/pending/Kiosk-0.1.89-armv7l.AppImage
[2021-10-21 13:44:15.741] [info] Electron App: AutoUpdater update-downloaded. version - 0.1.89
[2021-10-21 13:44:15.743] [info] Electron App: Install on explicit quitAndInstall
[2021-10-21 13:44:15.745] [info] Electron App: Install: isSilent: true, isForceRunAfter: true
[2021-10-21 13:44:15.978] [info] Electron App: Quitting...
[2021-10-21 13:44:15.979] [info] Electron App: Update installer has already been triggered. Quitting application.
[2021-10-21 13:44:19.357] [info] Electron App: Starting...
[2021-10-21 13:44:19.366] [info] Electron App: Platform - "linux"
[2021-10-21 13:44:19.367] [info] Electron App: Arch - "arm"
[2021-10-21 13:44:19.368] [info] Electron App: App version - "0.1.89"
[2021-10-21 13:44:19.370] [info] Electron App: CMD arguments - {"_":[]}
[2021-10-21 13:44:20.156] [info] Electron App: App ready
[2021-10-21 13:44:20.158] [info] Electron App: AutoUpdate Channel - "latest"
[2021-10-21 13:44:20.655] [info] Electron App: sudo sed -n "s/disable_overscan=//p" /boot/config.txt
[2021-10-21 13:44:20.789] [info] Electron App: 1

[2021-10-21 13:44:21.173] [info] Electron App: Successfully connected to internet
[2021-10-21 13:44:21.189] [info] Electron App: Checking for update
[2021-10-21 13:44:21.783] [info] Electron App: Update for version 0.1.89 is not available (latest version: 0.1.89, downgrade is allowed).
[2021-10-21 13:44:21.784] [debug] Electron App: checkForUpdatesAndNotify called, downloadPromise is null
[2021-10-21 13:44:21.786] [info] Electron App: Current ansible version: undefined
[2021-10-21 13:44:21.787] [info] Electron App: Running ansible update 0.1.89
[2021-10-21 13:44:21.809] [info] Electron App: ansible-playbook /tmp/.mount_kiosk-UvUw24/resources/ansible/update.yml 2>> /home/pi/ansible.log 1>> /home/pi/ansible.log
[2021-10-21 13:44:21.842] [error] Electron App: Error: Command failed: ansible-playbook /tmp/.mount_kiosk-UvUw24/resources/ansible/update.yml 2>> /home/pi/ansible.log 1>> /home/pi/ansible.log

[2021-10-21 13:44:21.869] [error] Electron App: Error: Command failed: ansible-playbook /tmp/.mount_kiosk-UvUw24/resources/ansible/update.yml 2>> /home/pi/ansible.log 1>> /home/pi/ansible.log

**Comment by igolus** [comment]
Inside H:\home\pi\ansible.log

/bin/sh: 1: ansible-playbook: not found

**Comment by igolus** [comment]
I tried to copy /usr/bin/ansible-playbook from another card where i am able to connect via SSH using ext2. But the file does not persists in the file system !!!

No solution found yet :(

**Comment by finnie2006** [comment]
is the project dead

**Comment by fitchsm** [comment]
I am having the same issue too, suck in update on first start up

**Comment by fitchsm** [comment]
Found a work around!! Using a Raspberry Pi 3B + and Raspberry OS with desktop
Follow these steps:

1) Using Raspberry Pi Imager
     a. Option to use SSH, really helpful if you need to do any changes after deployment. Ctrl + Shift + X for advanced options enable SSH and set password and set locale too. Click save
     b. Using Imager, install on SD card the latest Raspberry OS with Desktop (Yes, you need the Desktop)

2) Install SD into Raspberry Pi

3) Once boot up is complete, using the terminal (or SSH into with putty, login as user "pi" and password)

4) In terminal input this command: $ sudo raspi-config 
     a. Go to 4 Performance Options, P2 GPU Memory
     b. Set GPU memory to 256mb
     c. Exit allow for reboot 

5) Things to do in the desktop
     a. Disable screen blanking with Raspberry Pi Configuration Window under Display Tab
     b. Optional but highly recommended that you disable Bluetooth to prevent hacking
     c. Do any updates now if indicated

6) Log back in to terminal  input this command:  $  sudo nano /etc/xdg/lxsession/LXDE-pi/autostart
     a. At end of file, input this command for the autostart of chromium in Kiosk mode:
chromium-browser --kiosk https://tv.kiosk.show/   
     b. Ctrl x, select y for yes to write, and hit enter without changing the file name
     c. input this command:  $ sudo reboot

7) With first reboot, screen will be a dark blue but will eventually populate with the Kiosk paring screen. I noticed that on second and beyond reboots, the screen populates much faster

Congrads, you are set up with Kiosk on Raspberry Pi using Chomium in kiosk mode

Let me know how this works for you. I am interested to knowing the performance of this with a RP4

Steve

**Comment by finnie2006** [comment]
That should work

**Comment by finnie2006** [comment]
Maybe you can try this
https://github.com/kihosk/kiosk-os

**Comment by Bebreizh** [comment]
> Maybe you can try this https://github.com/kihosk/kiosk-os

Where did you write those command lines ?

Same issue on Rpi 4b 8Go, with both 14 and 15 versions of the project.

**Comment by Azureus007** [comment]
I am having the same issue too, stuck in update. Anyone help?

**Comment by vlChris** [comment]
![IMG_3062](https://user-images.githubusercontent.com/12664606/145682051-82a41428-f855-48c2-b6dc-bab2fcfd38c1.jpg)

Sigh.. 

I guess this is 30 minutes of my life I won't get back. Too bad, this project actually looked really cool.

**Comment by Postiopolis** [comment]
I switched to PiSignage and it's been working great and has an interface that I can use on my computer remotely to change the displays.

**Comment by adanguillen** [comment]
Hello, I have had the same problem, the solution has been to access the terminal with **CTRL+ALT+F4**, log in and execute 2 commands:
**- sudo apt update**
**- sudo apt upgrade**
I realized that the process takes a long time, so I suppose that the dependencies are quite outdated, at the end of a simple
**- sudo reboot**
and everything started to work, we are currently using it in my company for internal news on the TV in the cafeteria and it works very well
**Raspberry Pi 4 8GB has been used**

**Comment by Jindra-Dev04** [comment]
> Hello, I have had the same problem, the solution has been to access the terminal with **CTRL+ALT+F4**, log in and execute 2 commands:
> **- sudo apt update**
> **- sudo apt upgrade**
> I realized that the process takes a long time, so I suppose that the dependencies are quite outdated, at the end of a simple
> **- sudo reboot**
> and everything started to work, we are currently using it in my company for internal news on the TV in the cafeteria and it works very well
> **Raspberry Pi 4 8GB has been used**

What are the login credentials?

**Comment by adanguillen** [comment]
> > Hello, I have had the same problem, the solution has been to access the terminal with **CTRL+ALT+F4**, log in and execute 2 commands:
> > **- sudo apt update**
> > **- sudo apt upgrade**
> > I realized that the process takes a long time, so I suppose that the dependencies are quite outdated, at the end of a simple
> > **- sudo reboot**
> > and everything started to work, we are currently using it in my company for internal news on the TV in the cafeteria and it works very well
> > **Raspberry Pi 4 8GB has been used**
> 
> What are the login credentials?

Good, when flashing with the **Raspberry Pi Imager**, in advanced options, when **enabling ssh**, it lets you create a user with a password, the user has to be "**pi**" because the software routes are **configured for "pi" user**, the password it is your own choice.

https://www.raspberrypi.com/software/

**Comment by SSTT3** [comment]
stuck on updating hardware?..SOLUTION...when writing image on raspberry imager activate SSH and write your own password then mount it on raspberry pi then wait for the screen Updating hardware then  CTRL+ALT+F4, login_PI and your pasword next on terminal-- sudo apt update-after  sudo apt upgrade-after-select n  for default. then wait to finish then type -reboot....its work for my raspberry pi 4

**Comment by caprisun08** [comment]
> Found a work around!! Using a Raspberry Pi 3B + and Raspberry OS with desktop Follow these steps:
> 
> 1. Using Raspberry Pi Imager
>    a. Option to use SSH, really helpful if you need to do any changes after deployment. Ctrl + Shift + X for advanced options enable SSH and set password and set locale too. Click save
>    b. Using Imager, install on SD card the latest Raspberry OS with Desktop (Yes, you need the Desktop)
> 2. Install SD into Raspberry Pi
> 3. Once boot up is complete, using the terminal (or SSH into with putty, login as user "pi" and password)
> 4. In terminal input this command: $ sudo raspi-config
>    a. Go to 4 Performance Options, P2 GPU Memory
>    b. Set GPU memory to 256mb
>    c. Exit allow for reboot
> 5. Things to do in the desktop
>    a. Disable screen blanking with Raspberry Pi Configuration Window under Display Tab
>    b. Optional but highly recommended that you disable Bluetooth to prevent hacking
>    c. Do any updates now if indicated
> 6. Log back in to terminal  input this command:  $  sudo nano /etc/xdg/lxsession/LXDE-pi/autostart
>    a. At end of file, input this command for the autostart of chromium in Kiosk mode:
>    chromium-browser --kiosk https://tv.kiosk.show/
>    b. Ctrl x, select y for yes to write, and hit enter without changing the file name
>    c. input this command:  $ sudo reboot
> 7. With first reboot, screen will be a dark blue but will eventually populate with the Kiosk paring screen. I noticed that on second and beyond reboots, the screen populates much faster
> 
> Congrads, you are set up with Kiosk on Raspberry Pi using Chomium in kiosk mode
> 
> Let me know how this works for you. I am interested to knowing the performance of this with a RP4
> 
> Steve

Yes, this works running on pi3 B+ ( I was also stuck on the update screen with running the standard newest image)

**Comment by imogenbertin** [comment]
"Found a work around!! Using a Raspberry Pi 3B + and Raspberry OS with desktop"
Thank you so much. I'm not entirely sure if it means I'm more gaga or less than I thought but I got a community digital sign working again on pi3 B+ using Steve/fitchsm's workaround including remembering how to use vi which I think I last used in 1982. The sign's been off since lockdown so great to have it back again.

**Comment by karibi05** [comment]
I tried to follow your solutions, however for me this worked only:
sudo apt-get update --allow-releaseinfo-change
and
sudo apt-get upgrade should do it

more reference:  https://hakanu.net/linux/2022/07/03/resolve-apt-update-problem-on-raspberry-pi-this-must-be-accepted-explicitly-before-updates-for-this-repository/

**Comment by caprisun08** [comment]
> > Found a work around!! Using a Raspberry Pi 3B + and Raspberry OS with desktop Follow these steps:
> > 
> > 1. Using Raspberry Pi Imager
> >    a. Option to use SSH, really helpful if you need to do any changes after deployment. Ctrl + Shift + X for advanced options enable SSH and set password and set locale too. Click save
> >    b. Using Imager, install on SD card the latest Raspberry OS with Desktop (Yes, you need the Desktop)
> > 2. Install SD into Raspberry Pi
> > 3. Once boot up is complete, using the terminal (or SSH into with putty, login as user "pi" and password)
> > 4. In terminal input this command: $ sudo raspi-config
> >    a. Go to 4 Performance Options, P2 GPU Memory
> >    b. Set GPU memory to 256mb
> >    c. Exit allow for reboot
> > 5. Things to do in the desktop
> >    a. Disable screen blanking with Raspberry Pi Configuration Window under Display Tab
> >    b. Optional but highly recommended that you disable Bluetooth to prevent hacking
> >    c. Do any updates now if indicated
> > 6. Log back in to terminal  input this command:  $  sudo nano /etc/xdg/lxsession/LXDE-pi/autostart
> >    a. At end of file, input this command for the autostart of chromium in Kiosk mode:
> >    chromium-browser --kiosk https://tv.kiosk.show/
> >    b. Ctrl x, select y for yes to write, and hit enter without changing the file name
> >    c. input this command:  $ sudo reboot
> > 7. With first reboot, screen will be a dark blue but will eventually populate with the Kiosk paring screen. I noticed that on second and beyond reboots, the screen populates much faster
> > 
> > Congrads, you are set up with Kiosk on Raspberry Pi using Chomium in kiosk mode
> > Let me know how this works for you. I am interested to knowing the performance of this with a RP4
> > Steve
> 
> Yes, this works running on pi3 B+ ( I was also stuck on the update screen with running the standard newest image)

Ran this again on Rapsberry Pi 4 model b (4gb ram) to have two kiosks in the building, this method works really good
